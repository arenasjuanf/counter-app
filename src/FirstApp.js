import PropTypes from "prop-types";

const FirstApp = ({ name = "default" }) => {
    const greetings =  "Hello World";

    return (
        <>
            <h1> { greetings } </h1>
            <h5>- {name} -</h5>
        </>
    );
}

FirstApp.propTypes = {
    name: PropTypes.string, // .isRequired,
}

FirstApp.defaultProps = {
    subtitle : "example subtitle"
}

export default FirstApp;