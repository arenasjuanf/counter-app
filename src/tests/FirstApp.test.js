import '@testing-library/jest-dom/extend-expect';
import FirstApp from '../FirstApp';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';
import {createSerializer} from 'enzyme-to-json';


import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

describe("Pruebas en FirstApp", () => {

   /*  test("should show  'Juanf'", ()=>{

        const name = 'Juanf';
        const wrapper = render(<FirstApp name={ name } />);
        expect( wrapper.getByText(name) ).toBeInTheDocument();
    }) */

    test("should show <FirstApp/>", () => {
        expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

        
        const wrapper = shallow(<FirstApp name="Juanf" />);
        expect(wrapper).toMatchSnapshot();

    })

    test("should show 'name' propertie sended by props ", () => {
        expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

        const wrapper = shallow(<FirstApp name="Juanf" />);
        const wrapperText = wrapper.find("h5").text();
        expect(wrapperText).toBe("Juanf");

    })

})