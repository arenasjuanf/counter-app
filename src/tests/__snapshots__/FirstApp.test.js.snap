// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp should show <FirstApp/> 1`] = `
<Fragment>
  <h1>
     
    Hello World
     
  </h1>
  <h5>
    Juanf
  </h5>
</Fragment>
`;
